### DISCUSSION TITLE
Map Rotation - Automatic Zoom - Backup Quick Actions

### DISCUSSION THREAD

**Discussion by peterelli** [osmandapp/OsmAnd#17796] [Ideas]
Hello all,

1. when I follow my track the map is always rotated according to my direction of movement, great. But when I stop at an intersection, the map rotates in some other direction which makes it very hard to orient. Previously, the map stopped in this direction of approach so that the map view matched the reality of the intersection.

2) I like the function of the speed-dependent map zoom. Unfortunately, since the last updates, the zoom jumps back and forth much too fast when you are driving in the speed range in which the zoom switching takes place. Please make the zoom switching more sluggish and expand the possibility in the settings to be able to enter speeds and associated zoom values themselves.

3. is it possible to transfer the settings quick actions to another device?

Thanks for the regular development
All the best Peter

**Comment by vshcherb** [comment]
Indeed we're going to review autozoom / rotate algorithm completely in 4.6 cause our algorithm was mostly focused in 2D and now it needs to change

**Reply by deprado63** [reply]
Vor allem Punkt 1 ist sehr wichtig Da man wirklich nicht mehr weiß wohin soll ich fahren Also unbedingt beheben.

**Reply by sjvudp** [reply]
I wonder: How hard would it be to implement a "speed:zoom_factor" profile, that is specify speed steps and zoom steps, maybe by drawing a simple graph?
Another idea is to make the zoom factor depend on the turns and crossings ahead (the closer the next turn/crossing (or turn/crossing behind) the more detailed the map). So inside towns one would see many details, while on cross-country one would see less details, even when driving fast.

**Comment by TobiPeterG** [comment]
Hi there,
I'm on 4.8.5 from fdroid.
I personally feel like the auto zoom should be further away from the current position, I always feel like it should be two to three zoom factors further away (I'm using 2.5/3D view)

Also, I find it confusing that at intersections the view rotates before I make the turn.

**Reply by DmitryAlexei** [reply]
Auto-zoom level can be configured in the settings. See docs https://osmand.net/docs/user/navigation/guidance/map-during-navigation#map-during-navigation

we have an issue with the premature rotation of the map view https://github.com/osmandapp/OsmAnd/issues/20422

**Reply by TobiPeterG** [reply]
Hey, thanks for the answer :)

I might be blind though 😅

https://youtu.be/T_Z1cmdVSXo

**Reply by TobiPeterG** [reply]
> Auto-zoom level can be configured in the settings. See docs https://osmand.net/docs/user/navigation/guidance/map-during-navigation#map-during-navigation
> 
> we have an issue with the premature rotation of the map view https://github.com/osmandapp/OsmAnd/issues/20422

Oh my god, I am really blind, the menu entry has no icon so I thought it was some explanatory text. -_-
Sorry
